### ISSUE TITLE
not all c style formatting specifiers are detected

### ISSUE THREAD

**Issue by lcnr** [rust-lang/rust#89173]
i've tried
```rust
fn main() {
    let num = 0x0abcde;
    let width = 6;
    print!("%0*x", width, num);
}
```
i expected:
```
error: multiple unused formatting arguments
 --> src/main.rs:4:19
  |
4 |     print!("%0*x", width, num);
  |            -----  ^^^^^  ^^^ argument never used
  |            ||     |
  |            ||     argument never used
  |            |help: format specifiers use curly braces: `{1:00$x}`
  |            multiple missing formatting specifiers
  |
  = note: printf formatting not supported; see the documentation for `std::fmt`
```
the actual error message does not contain such a help message:
```
error: multiple unused formatting arguments
 --> src/main.rs:4:20
  |
4 |     print!("%0*x", width, num);
  |            ------  ^^^^^  ^^^ argument never used
  |            |       |
  |            |       argument never used
  |            multiple missing formatting specifiers
```

**Comment by lcnr** [comment]
note that the following does suggest `{:06x}`
```rust
fn main() {
    let num = 0x0abcde;
    print!("%06x", num);
}
```

**Comment by jamesmunns** [comment]
Is there a typo in your snippet?

```rust
print!("%0*x", width, num);
```

The `*` looks out of place to me, even for printf formatting. It's also not matched in your "expected" output (the `*` is not there).

**Comment by lcnr** [comment]
no, `*` is used to get the width specifier from the arguments, which is why the equivalent rust specifier would be `{1:00$x}` here. see https://rust.godbolt.org/z/MvdsPTWTa for working `c` code

**Comment by jamesmunns** [comment]
Wow! Thanks for explaining, TIL (in both Rust and C!)
